scripts/h5peditor-pre-save.js

```diff
@@ -0,0 +1,156 @@
+var H5PEditor = H5PEditor || {};
+var H5PPresave = H5PPresave || {};
+
+H5PEditor.Presave = (function (Editor) {
+  "use strict";
+
+  /**
+   * Presave structure
+   *
+   * @class
+   */
+
+  function Presave() {
+    this.maxScore = 0;
+  }
+
+  /**
+   * Process the given library and calculate the max score
+   *
+   * @public
+   * @param {string} library
+   * @param {object} content
+   * @returns {H5PEditor.Presave}
+   */
+  Presave.prototype.process = function (library, content) {
+    var self = this;
+
+    library = Presave.sanitizeLibrary(library);
+    if (Presave.libraryExists(library) === true) {
+      H5PPresave[library](content, function (serverSideData) {
+        if (typeof serverSideData !== 'object') {
+          return;
+        }
+        if (serverSideData.hasOwnProperty('maxScore') && Presave.isInt(serverSideData.maxScore)) {
+          self.maxScore += serverSideData.maxScore;
+        }
+      });
+    }
+    return this;
+  };
+
+  /**
+   * Check if the score is valid or throw exception if not
+   *
+   * @static
+   * @param score
+   * @returns {boolean}
+   * @throws {Presave.exceptions.InvalidMaxScoreException} If score is not valid
+   */
+  Presave.validateScore = function (score) {
+    if (!Presave.isInt(score) || score < 0) {
+      throw new this.exceptions.InvalidMaxScoreException();
+    }
+    return true;
+  };
+
+  /**
+   * Check if score is valid
+   *
+   * @static
+   * @param score
+   * @returns {boolean}
+   */
+  Presave.isScoreValid = function (score) {
```

Comment:
This function is never used. Is this intended ?
 Exception is thrown if max score is 0 ref: https://h5ptechnology.atlassian.net/browse/HFP-1807. Not sure if this should be an acceptance criteria though, as there are many times where a content type would return 0 maxscore, e.g. CP without any scorable interactions.

Reply:
As I remember I used the method in an early version of the code and left it there for convenience purposes. But I can remove it as it's not currently being used.